lib/turkish_support/helpers.rb

```diff
@@ -0,0 +1,112 @@
+module TurkishSupportHelpers
+  def translate_regexp(pattern)
+    Regexp.new(pattern) unless pattern.is_a? Regexp
+    re, options = pattern.source, pattern.options
+
+    while re.match(RANGE_REGEXP)
+      re.scan(RANGE_REGEXP).flatten.compact.each do |matching|
+        re.gsub! matching, translate_range(matching, pattern.casefold?)
+      end
+    end
+
+    META_CHARS.each { |k, v| re.gsub!(k, v) }
+    Regexp.new(re.force_encoding('UTF-8'), Regexp::FIXEDENCODING | options)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.